public/editor/scripts/project-list.js

```diff
@@ -81,44 +81,88 @@ require(["jquery", "constants", "analytics", "moment"], function($, Constants, a
+            err = err || new Error("unknown network error");
+            console.error(err);
+          });
+
+          project.hide({
+            duration: 250,
+            easing: "linear",
+            done: function() {
+              project.remove();
+            }
+          });
+        });
+        projectsToDelete = [];
+        $(".delete-button").remove();
+      };
+      var listWrapper = document.getElementsByClassName("project-list-wrapper");
+      listWrapper[0].appendChild(deleteBtn);
     }
 
     var project = $(this).closest(".project");
-    var projectId = project.attr("data-project-id");
-    var projectElementId = project.attr("id");
-    $("#" + projectElementId + " > .project-title").off("click");
-
-    analytics.event({ category : analytics.eventCategories.PROJECT_ACTIONS, action : "Delete Project" });
-
-    var request = $.ajax({
-      headers: {
-        "X-Csrf-Token": $("meta[name='csrf-token']").attr("content")
-      },
-      type: "DELETE",
-      url: "/" + locale + "/projects/" + projectId,
-      timeout: Constants.AJAX_DEFAULT_TIMEOUT_MS
-    });
-    request.done(function() {
-      if(request.status !== 204) {
-        console.error("[Thimble error] sending delete request for project ", projectId, request.status);
+    var projectSelector = "#" + project.attr("id");
+    var deleteIndex = projectsToDelete.findIndex(function(x) {return x.attr("id") === project.attr("id");});
+    var disabledLink = {"cursor":"default", "text-decoration":"none"};
+    var enabledLink = {"cursor":"pointer", "text-decoration":"underline"};
+    
+    if(deleteIndex === -1) {
+      projectsToDelete.push(project);
+      $(projectSelector + " .edit-link").bind("click", false).css({"cursor":"default"});
+      if($(projectSelector + " .published-link").length !== 0) {
+        $(projectSelector + " .published-link").bind("click", false).css(disabledLink);
+        $(projectSelector + " .remix-link").bind("click", false).css(disabledLink);
       }
-    });
-    request.fail(function(jqXHR, status, err) {
-      err = err || new Error("unknown network error");
-      console.error(err);
-    });
-
-    project.hide({
-      duration: 250,
-      easing: "linear",
-      done: function() {
-        project.remove();
+      project.css("background", "rgba(255,0,0,.08)");
+      $(this).closest(".project-delete").html("Cancel");
+    } else {
+      projectsToDelete.splice(deleteIndex, 1);
+      $(projectSelector + " .edit-link").unbind("click", false).css({"cursor":"pointer"});
+      if($(projectSelector + " .published-link").length !== 0) {
+        $(projectSelector + " .published-link").unbind("click", false).css(enabledLink);
+        $(projectSelector + " .remix-link").unbind("click", false).css(enabledLink);
       }
-    });
+      project.css("background", "");
+      $(this).closest(".project-delete").html("<div class='icon-garbage-can'></div> {{prjListDeleteProjectBtn}}");
```

Comment:
`${this}.closest` is setting off some alarm bells: we should know which element we're putting this HTML code into. However, the bigger alarm bell is that this explicitly bakes a security exploit into our code: if a localization intentionally contains script code as part of the `prjListDeleteProjectBtn` translation then that code *will run*. Instead, Create the div, add it, then create a span with the locale string as `text`, *not* as `html`, and then add that.